hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -124,11 +123,7 @@ public ClientMessage getClientMessage() {
 
     public ClientInvocationFuture invoke() {
         assert (clientMessage != null);
-        try {
-            clientMessage.setCorrelationId(callIdSequence.next());
-        } catch (TimeoutException e) {
-            throw new HazelcastOverloadException("Timed out trying to acquire another call ID.", e);
-        }
+        clientMessage.setCorrelationId(callIdSequence.next());
```

Comment:
In the future the invocation needs to have a proper toString explaining the type of operation being invocation; just as is done on the server. Once that is added; we need to restore the try/catch around it and it will be equally strange as on the server where you catch the exception and then rethrow an exception of the same type.

Reply:
Yes, I agree, the client and server side should work as similar as possible. Since it is not used at this time, may be we should add it in another PR where this invocation is being logged. Without that piece of the code, it will just mean some more code without coverage.